core/lib/src/fs/server.rs

```diff
@@ -94,6 +94,20 @@ impl FileServer {
             .rewrite(DirIndex::unconditional("index.html"))
     }
 
+    /// Constructs a new `FileServer` that serves files from the file system
+    /// `path` with a default rank.
+    ///
+    /// Adds a set of default rewrites:
+    /// - [`filter_dotfiles`]: Hides all dotfiles.
+    /// - [`prefix(path)`](prefix): Applies the root path.
+    /// - [`normalize_dirs`]: Normalizes directories to have a trailing slash.
+    pub fn directory<P: AsRef<Path>>(path: P) -> Self {
+        Self::empty()
+            .filter(|f, _| f.is_visible())
+            .rewrite(Prefix::checked(path))
+            .rewrite(TrailingDirs)
+    }
```

Comment:
So I had implemented this exact method but I removed it because I doubted its utility, given that this would make `new()` exactly `Self::directory(path).rewrite(DirIndex::unconditional("index.html"))`. If we're going to keep this, we should document the two methods well with why/when you would use one or the other.

Reply:
I assumed something like that might have happened. It was still used in one of the doc-tests (of `DirIndex` specifically). I think it's worth including, but I'm thinking it might be worth changing the name in addition to writing better docs.